### PR TITLE
Just waiting 3 seconds when stop container.

### PR DESCRIPTION
**Pull request by hanpfei**
If container manager dies before session manager dying,
the session manager cannot exit if it waits for all.

Signed-off-by: Han Pengfei <hanpfei@gmail.com>

### CODE REVIEW

File: src/anbox/qemu/adb_message_processor.cpp [hunk 1]
```diff
@@ -74,10 +74,11 @@ void AdbMessageProcessor::advance_state() {
 
       if (state_ == closed_by_host) {
         host_connector_.reset();
-        return;
+      } else {
+        wait_for_host_connection();
       }
 
-      wait_for_host_connection();
+      lock_.unlock();
```
Comment (morphis) [inline src/anbox/qemu/adb_message_processor.cpp]: What is this actually fixing? That lock_ is kept lock is intentional.

File: src/anbox/qemu/adb_message_processor.cpp [hunk 2]
```diff
@@ -74,10 +74,11 @@ void AdbMessageProcessor::advance_state() {
 
       if (state_ == closed_by_host) {
         host_connector_.reset();
-        return;
+      } else {
```
Comment (morphis) [inline src/anbox/qemu/adb_message_processor.cpp]: Drop brackets

File: src/anbox/container/management_api_stub.h [hunk 3]
```diff
@@ -56,6 +56,7 @@ class ManagementApiStub : public DoNotCopyOrMove {
 
   mutable std::mutex mutex_;
   std::shared_ptr<rpc::Channel> channel_;
+  static const std::chrono::milliseconds stop_waiting_millis;
```
Comment (morphis) [inline src/anbox/container/management_api_stub.h]: This has to go into a anonymous namespace inside management_api_stub.cpp

Furthermore the variable type is already encoding that we have milliseconds here so no need to repeat that in the variable name. stop_waiting_timeout is what you want here as a name

File: src/anbox/container/management_api_stub.cpp [hunk 4]
```diff
@@ -75,7 +78,7 @@ void ManagementApiStub::stop_container() {
   channel_->call_method("stop_container", &message, c->response.get(),
       google::protobuf::NewCallback(this, &ManagementApiStub::container_stopped, c.get()));
 
-  c->wh.wait_for_all();
+  c->wh.wait_for_pending(stop_waiting_millis);
```
Comment (morphis) [inline src/anbox/container/management_api_stub.cpp]: I know why but this deserves a comment in the code

File: src/anbox/qemu/adb_message_processor.cpp (hunk 1 above)
Comment (hanpfei) [reply src/anbox/qemu/adb_message_processor.cpp]: I found that sometimes the process will be blocked here, so it cannot exit normally.

File: src/anbox/qemu/adb_message_processor.cpp [hunk 5]
```diff
@@ -72,12 +72,12 @@ void AdbMessageProcessor::advance_state() {
...
-      if (state_ == closed_by_host) {
+      if (state_ == closed_by_host)
         host_connector_.reset();
-        return;
-      }
+      else
+        wait_for_host_connection();
 
-      wait_for_host_connection();
+      lock_.unlock();
```
Comment (morphis) [inline src/anbox/qemu/adb_message_processor.cpp]: As described before, this is still wrong.

**Pull request merged**